staging/src/k8s.io/apimachinery/pkg/runtime/serializer/json/json.go

```diff
@@ -242,6 +245,15 @@ func (s *Serializer) doEncode(obj runtime.Object, w io.Writer) error {
 		_, err = w.Write(data)
 		return err
 	}
+	if s.options.StreamingCollectionsEncoding {
+		ok, err := streamingEncode(obj, w)
+		if err != nil {
+			return err
+		}
+		if ok {
+			return nil
+		}
```

Comment:
Not sure if could simplify to
```suggestion
		if ok || err != nil {
			return err
		}
		// Fallback to a standard (non-streaming) encode if the obj isn't supported in `streamingEncode`.
```
Do we need to add a comment here and mention it would fallback to non-streaming encode?